mysql-test/service-test.sh

```diff
@@ -0,0 +1,228 @@
+service mysql stop
+
+# Hide commands for the rest of the script make output more readable
+set +x
+
+echo "Status after installation.."
+check_service_status "sysvinit=1;systemd=1;"
+echo
+
+echo "Disabling service in rc.d..."
+update-rc.d mariadb disable
+check_service_status "sysvinit=0;systemd=0;"
+echo
+
+echo "Enabling service in rc.d..."
+update-rc.d mariadb enable
+check_service_status "sysvinit=1;systemd=1;"
+echo
+
+echo "Disabling service in systemd..."
+systemctl disable mariadb
+check_service_status "sysvinit=0;systemd=0;"
+echo
+
+echo "Enabling service in systemd..."
+systemctl enable mariadb
+check_service_status "sysvinit=1;systemd=1;"
+echo
+
+echo "Disabling service in systemd with name 'mysql', not synced..."
+systemctl disable mysql
+check_service_status "sysvinit=1;systemd=0;"
+echo
+
+echo "Enabling service in systemd with name 'mysql'..."
+systemctl enable mysql
+# @TODO: this adds /etc/rc.*/S01mysql symlinks
+check_service_status "sysvinit=1;systemd=1;"
+echo
+
+echo "Masking service in systemd..."
+systemctl mask mariadb
+check_service_status "masked=1;"
+
+echo "Unmasking service in systemd..."
+systemctl unmask mariadb
+check_service_status "sysvinit=1;systemd=1;"
+
+echo "Disabling service in systemd..."
+systemctl disable mariadb
+check_service_status "sysvinit=0;systemd=0;"
+echo
+
+echo "Enabling service in systemd..."
+systemctl enable mariadb
+check_service_status "sysvinit=1;systemd=1;"
+echo
+
+# @TODO: There is no /etc/init.d/msyqld so disabling mysqld in systemd with
+# 'systemctl disable mysqld' does not trigger /lib/systemd/systemd-sysv-install
+# for service name 'mysqld' and thus disabling the sysvinit service mariadb service
+# stays enabled:
+#   Synchronizing state of mariadb.service with SysV service script with /lib/systemd/systemd-sysv-install.
+#   Executing: /lib/systemd/systemd-sysv-install disable mariadb
+
+echo "Disabling service in systemd..."
+systemctl disable mysqld
+check_service_status "sysvinit=1;systemd=0;"
+echo
+
+echo "Enabling service in systemd..."
+systemctl enable mysqld
+check_service_status "sysvinit=1;systemd=1;"
+echo
+
+# NOTE! Script /usr/sbin/service is just a wrapper for init or systemctl,
+# no need to test it separately. While invoke-rc.d is primairly a sysvinit tool,
+# it also runs systemctl commands under the hood if systemd is detected.
+
+# @TODO Test also service dependencies (Requires/Wants)
+# https://www.freedesktop.org/software/systemd/man/systemd.unit.html#Wants=
+cat > /lib/systemd/system/mariadbdependentservice.service <<EOF
+[Unit]
+Requires=mariadb.service
+
+[Service]
+ExecStart=/bin/sleep 3600
+EOF
+
+cat > /lib/systemd/system/mysqldependentservice.service <<EOF
+[Unit]
+Requires=mysql.service
+
+[Service]
+ExecStart=/bin/sleep 3600
+EOF
+
+systemctl daemon-reload
+systemctl stop mariadb
+sleep 5
```

Comment:
`while systemctl status mariadb; do sleep 2; done` as alternate to sleep

And thanks for including these tests.

Reply:
I don't plan to further work on this test anymore. It served it's role in my local testing and until there is a facility that could run it, there is not much point in fine tuning it more.